_docs/tutorials/event-sourcing-and-cqrs-modeling.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Event Sourcing and CQRS Modeling in Context Mapper
+permalink: /docs/event-sourcing-and-cqrs-modeling/
+---
+
+Event sourcing and Command Query Responsibility Segregation (CQRS) are two different approaches, but they work very well together. Both concepts deal with application state.
+Event sourcing captures all changes to the state as a sequence of (or stream) of events so that state changes can be communicated flexibly in loosely coupled systems. This is why this 
+concept is interesting for microservice architectures. CQRS separates query processing from the create, update, delete business logic so that different Non-Functional Requirements (NFRs), for instance regarding performance, for read and write access can be satisfied in different ways. In complex domains, this is often desired. 
+
+If you are not fully familiar with the concepts of event sourcing and CQRS yet, we recommend the following links:
+
+ * [Event sourcing](https://martinfowler.com/eaaDev/EventSourcing.html) by Martin Fowler
+ * [Event sourcing pattern](https://microservices.io/patterns/data/event-sourcing.html) by Chris Richardson
+ * [Event sourcing workshop slides](https://speakerdeck.com/mploed/event-sourcing-workshop-at-software-architecture-summit-2016) by Michael Plöd
+ * [Command query separation](https://martinfowler.com/bliki/CommandQuerySeparation.html) and [CQRS](https://martinfowler.com/bliki/CQRS.html) by Martin Fowler
+ * [Developing Transactional Microservices Using Aggregates, Event Sourcing and CQRS](https://www.infoq.com/articles/microservices-aggregates-events-cqrs-part-1-richardson/) by Chris Richardson
+ * [Designing Event Sourced Microservices](https://www.infoq.com/news/2017/11/event-sourcing-microservices/) by Jan Stenberg
+ 
+## Modeling in Context Mapper
+This page highlights the [Context Mapper DSL (CML)](/docs/language-reference/) concepts that support modeling event sourced systems and CQRS. 
+
+Within CML Bounded Contexts and Aggregates we integrated the [Sculptor DSL](http://sculptorgenerator.org/) that allows to specify the domain model of the Bounded Contexts.
+
+### Events
+As explained in the [Sculptor documentation](http://sculptorgenerator.org/documentation/event-driven-tutorial), the tactic DDD syntax of Sculptor supports modeling events.
+
+A domain event is something that happened that affects the application/resource state. [Michael Plöd](https://speakerdeck.com/mploed/event-sourcing-workshop-at-software-architecture-summit-2016)
+suggests that the scope of domain events is always based on Aggregates. The names of domain events shall indicate that the event happened in the past. 
+
+The following example illustrates how you can model domain events within your Aggregate: 
+<!-- Q: do the generators (puml, MDSL) already process event emission (publish) and reception (subscribe)? do the below example cover these things? see http://sculptorgenerator.org/documentation/event-driven-tutorial -->
```

Comment:
What do you mean exactly here @socadk? As I understand it, the publish/subscribe mechanisms here are used to generate messaging-specific code (message queues). Events can be modelled technology-independent, as parameters or return types of operations/methods. The generators process these parameters or return types, yes.

Reply:
Short answer: no.
As discussed we aim to create a first tutorial without changing anything in Context Mapper.